sheets/mail_api.py

```diff
@@ -1,21 +1,57 @@
 """Mail API for sheets app"""
+import logging
 from urllib.parse import urlencode
 from collections import namedtuple
 
 import requests
 from django.conf import settings
+from rest_framework import status
 
 from ecommerce.constants import BULK_ENROLLMENT_EMAIL_TAG
 from mail.constants import MAILGUN_API_DOMAIN
+from sheets.constants import MAILGUN_API_TIMEOUT_RETRIES
 from sheets.utils import format_datetime_for_mailgun
 from mitxpro.utils import has_all_keys
 
+
+log = logging.getLogger(__name__)
+
 BulkAssignmentMessage = namedtuple(
     "BulkAssignmentMessage",
     ["bulk_assignment_id", "coupon_code", "email", "event", "timestamp"],
 )
 
 
+def _make_mailgun_api_request(url):
```

Comment:
Take or leave, but there isn't much that is specific to mailgun here. Is it worth refactoring to make it more generic?

Reply:
Good call